torch/distributed/distributed_c10d.py

```diff
@@ -1714,6 +1715,9 @@ def broadcast_object_list(object_list, src=0, group=None):
         src (int): Source rank from which to broadcast ``object_list``.
         group: (ProcessGroup, optional): The process group to work on. If None,
             the default process group will be used. Default is ``None``.
+        dist_device (``torch.device``, optional): device to send from or receive
```

Comment:
Would be good to specify that if this is not `None`, then intermediate tensor representations of objects, or any tensors within the contained objects, will be moved to this device before broadcasting. 

Also mention that default for this arg is None.

Reply:
Good point, revised.